src/theme/mod.rs

```diff
@@ -205,16 +207,86 @@ where
 
 #[derive(PartialEq, Debug, Default)]
 struct ExtensionMappings {
-    mappings: Vec<(glob::Pattern, Style)>,
+    mappings: Vec<GlobPattern>,
+}
+
+#[derive(PartialEq, Debug)]
+enum GlobPattern {
+    Complex(glob::Pattern, Style),
+
+    // TODO: the String here should probably be an inline string, since we're likely working with
+    // small strings (file extensions).
+    Simple(HashMap<String, Style>),
 }
 
 impl ExtensionMappings {
+    // helper for test suite
+    #[cfg(all(test, unix))]
+    fn to_vec_pat_style(&self) -> Vec<(glob::Pattern, Style)> {
+        let mut out = Vec::new();
+        for map in &self.mappings {
+            match map {
+                GlobPattern::Complex(p, s) => {
+                    out.push((p.clone(), *s));
+                }
+                GlobPattern::Simple(h) => {
+                    let mut simple_pats = h
+                        .iter()
+                        .map(|(k, v)| (glob::Pattern::new(&format!("*.{}", k)).unwrap(), *v))
+                        .collect::<Vec<(glob::Pattern, Style)>>();
+
+                    simple_pats.sort_by_key(|x| x.0.clone());
+
+                    out.extend(simple_pats);
+                }
+            }
+        }
+        out
+    }
+
```

Comment:
I think this should be moved to the `#[cfg(test)]` block. You can repeat `impl ExtensionMappings {` there and put your test helper in it.